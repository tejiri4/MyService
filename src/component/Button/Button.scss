@import '../../assets/variable.scss';

.button-component {
  button {
    position: relative;
    padding: 5px 5px;
    font-weight: 700;
    font-size: 15px;
    border-radius: 5px;
    outline: 0;
    overflow:hidden;
    background: none;
    cursor: pointer;
    transition:         0.08s ease-in;
    -o-transition:      0.08s ease-in;
    -ms-transition:     0.08s ease-in;
    -moz-transition:    0.08s ease-in;
    -webkit-transition: 0.08s ease-in;
  }

  .fill:hover {
    color: #fff !important;
  }

  .fill:before {
    content: "";
    position: absolute;
    background-color: $Vivid-blue;
    bottom: 0;
    left: 0;
    right: 0;
    top: 100%;
    z-index: -1;
    transition: top 0.09s ease-in;
  }

  .fill:hover:before {
    top: 0;
  }
}
