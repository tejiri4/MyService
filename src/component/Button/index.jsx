// react libraries
import React from 'react';
import PropTypes from 'prop-types';

//styles
import './Button.scss';

const Button = props => {
  const { width, height, startColor, endColor, color, name } = props;
  const buttonstyles = {
    width,
    height,
    backgroundImage: `linear-gradient(to right, ${startColor} , ${endColor} )`,
    color
  };
  return (
    <div className='button-component'>
      <button className='button-component__button' style={buttonstyles}>{name}</button>
    </div>
  );
};

Button.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  startColor: PropTypes.string,
  endColor: PropTypes.string,
  color: PropTypes.string,
  name: PropTypes.string
};

export default Button;
