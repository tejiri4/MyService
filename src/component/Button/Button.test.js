// react libraries
import React from 'react';

// third part libraries
import { shallow } from 'enzyme';

// component
import Button from './';

describe('Button component', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      width: '100px',
      height: '40px',
      startColor: 'white',
      endColor: 'white',
      color: 'green'
    };
    wrapper = shallow(<Button {...props} />);
  });

  it('renders the button component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

