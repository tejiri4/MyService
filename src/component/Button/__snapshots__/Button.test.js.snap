// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the button component without crashing 1`] = `ShallowWrapper {}`;
